lib/datadog/tracing/contrib/aws/instrumentation.rb

```diff
@@ -60,6 +64,98 @@ def annotate!(span, context)
             span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_STATUS_CODE, context.safely(:status_code))
           end
 
+          def add_service_specific_tags(span, aws_service, params)
+            case aws_service
+            when 'sqs'
+              add_sqs_tags(span, params)
+            when 'sns'
+              add_sns_tags(span, params)
+            when 'dynamodb'
+              add_dynamodb_tags(span, params)
+            when 'kinesis'
+              add_kinesis_tags(span, params)
+            when 'eventbridge'
+              add_eventbridge_tags(span, params)
+            when 'states'
+              add_states_tags(span, params)
+            when 's3'
+              add_s3_tags(span, params)
+            end
+          end
+
+          def add_sqs_tags(span, params)
+            queue_url = params[:queue_url]
+            queue_name = params[:queue_name]
+            # Regular expression to match the SQS queue URL pattern
+            # https://sqs.sa-east-1.amazonaws.com/12345678/MyQueueName
+            pattern = %r{https://sqs\.[^/]+\.amazonaws\.com/(\d+)/([^/]+)}
+
+            if queue_url && (match = queue_url.match(pattern))
+              aws_account, queue_name = match.captures
+              span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
+            end
+            span.set_tag(Ext::TAG_QUEUE_NAME, queue_name)
+          end
+
+          def add_sns_tags(span, params)
+            topic_arn = params[:topic_arn]
+            topic_name = params[:name]
+            if topic_arn
+              # example topic_arn: arn:aws:sns:us-west-2:123456789012:my-topic-name
+              topic_name = topic_arn.split(':')[-1]
+              aws_account = topic_arn.split(':')[-2]
```

Comment:
Similar benchmark can be made here to show that it's better to split the string only once:
```ruby
require 'benchmark/ips'

topic_arn = 'arn:aws:sns:us-west-2:123456789012:my-topic-name'

Benchmark.ips do |x|
  x.time = 5
  x.warmup = 0.2

  x.report('two splits') do
    topic_name = topic_arn.split(':')[-1]
    aws_account = topic_arn.split(':')[-2]
  end

  x.report('one splits') do
    _,_,_,_,aws_account, topic_name = topic_arn.split(':')
  end

  x.compare!
end
```

```
Comparison:
          one splits:  1657116.8 i/s
          two splits:   873666.9 i/s - 1.90x  slower
```

Reply:
Yeah, double split could be avoided. Perhaps

```
irb(main):015:0> *_, a, b = [1,2,3,4]
=> [1, 2, 3, 4]
irb(main):016:0> a
=> 3
irb(main):017:0> b
=> 4
```